app/services/unpublish_handler_service.rb

```diff
@@ -0,0 +1,89 @@
+class UnpublishHandlerService
+  UNPUBLISH_LOG_PATH = "#{Rails.root}/log/unpublish_message.log".freeze
+
+  def initialize(content_id)
+    @content_id = content_id
+  end
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def call
+    if email_params.empty?
+      nil
+    elsif taxon_is_unpublished?
+      email_ids = UnpublishEmailBuilder.call(email_params).ids
+      log_taxon_emails(email_ids)
+    else
+      log_none_taxon_lists(subscriber_lists)
+    end
+  end
+
+private
+
+  attr_reader :content_id
+
+  def taxon_is_unpublished?
+    subscriber_lists.first.links.keys.include?(:taxon_tree)
+  end
+
+  def queue_delivery_request_workers(email_ids)
+    email_ids.each do |email_id|
+      DeliveryRequestWorker.perform_async_in_queue(
+        email_id, queue: :delivery_immediate
+      )
+    end
+  end
+
+  def email_params
+    subscriber_lists.flat_map do |subscriber_list|
+      subscriber_list.subscribers.activated.map do |subscriber|
+        {
+          subject: subscriber_list.title,
+          address: subscriber.address,
+          subscriber_id: subscriber.id
+        }
+      end
+    end
+  end
+
+  # For this query to return the content id has to be wrapped in a double quote blame psql 9.3
+  def subscriber_lists
+    @subscriber_lists ||= SubscriberList
+    .where(":id IN (SELECT json_array_elements((json_each(links)).value)::text)", id: "\"#{content_id}\"")
+    .includes(:subscribers).to_a
```

Comment:
This layout might be a little more readable.
```ruby
    @subscriber_lists ||= SubscriberList.where(
      ":id IN (SELECT json_array_elements((json_each(links)).value)::text)", id: "\"#{content_id}\""
    ).includes(
      :subscribers
    ).to_a
```

Or maybe just indent the lines after SubscriberList by two more spaces?

Reply:
I've indented and moved the `.to_a` to a separate line which i think looks the best!